integration-tests/test/sync-verifier.spec.ts

```diff
@@ -0,0 +1,149 @@
+      console.log(
+        `still equal`,
+        totalElementsBefore.toNumber(),
+        totalElementsAfter.toNumber()
+      )
+      totalElementsAfter = (await env.ctc.getTotalElements()) as BigNumber
+    }
+    return totalElementsAfter
+  }
+
+  const startAndSyncVerifier = async (sequencerBlockNumber: number) => {
+    // Bring up new verifier
+    verifier = new DockerComposeNetwork(['verifier'])
+    await verifier.up({ commandOptions: ['--scale', 'verifier=1'] })
+
+    // Wait for verifier to be looping
+    let logs = await verifier.logs()
+    while (!logs.out.includes('Starting Sequencer Loop')) {
+      console.log('Retrieving more logs')
+      await sleep(500)
+      logs = await verifier.logs()
+    }
+
+    const verifierProvider = injectL2Context(
+      new providers.JsonRpcProvider('http://localhost:8547')
+    )
+    console.log(await verifierProvider.getBlock('latest'))
+
+    // Wait until verifier has caught up to the sequencer
+    let latestVerifierBlock = (await verifierProvider.getBlock('latest')) as any
+    while (latestVerifierBlock.number < sequencerBlockNumber) {
+      console.log('waiting for new verifier blocks')
+      await sleep(500)
+      latestVerifierBlock = (await verifierProvider.getBlock('latest')) as any
+    }
+
+    return latestVerifierBlock
+  }
+
+  before(async () => {
+    env = await OptimismEnv.new()
+    wallet = env.l2Wallet
+  })
+
+  describe('Basic transactions', () => {
+    const initialAmount = 1000
+    const tokenName = 'OVM Test'
+    const tokenDecimals = 8
+    const TokenSymbol = 'OVM'
+
+    let other: Wallet
+    let Factory__ERC20: ContractFactory
+    let ERC20: Contract
+
+    before(async () => {
+      const env = await OptimismEnv.new()
+      wallet = env.l2Wallet
+      other = Wallet.createRandom().connect(ethers.provider)
+      Factory__ERC20 = await ethers.getContractFactory('ERC20', wallet)
+    })
+
+    afterEach(async () => {
+      await verifier.stop('verifier')
+      await verifier.rm()
+    })
+
+    it('should sync dummy transaction', async () => {
+      const totalElementsBefore = (await env.ctc.getTotalElements()) as BigNumber
+
+      const tx = {
+        to: '0x' + '1234'.repeat(10),
+        gasLimit: 4000000,
+        gasPrice: 0,
+        data: '0x',
+        value: 0,
+      }
+      const result = await wallet.sendTransaction(tx)
+      await result.wait()
+
+      const totalElementsAfter = await waitForBatchSubmission(
+        totalElementsBefore
+      )
+      expect(totalElementsAfter.gt(totalElementsAfter))
+
+      const latestSequencerBlock = (await provider.getBlock('latest')) as any
+      console.log(latestSequencerBlock)
+
+      const latestVerifierBlock = await startAndSyncVerifier(
+        latestSequencerBlock.number
+      )
+
+      expect(latestVerifierBlock.stateRoot).to.eq(
+        latestSequencerBlock.stateRoot
+      )
+    })
+
+    it('should sync ERC20 deployment and transfer', async () => {
```

Comment:
this test fails when it's deploy and transfer, also when it's only deploy without the transfer. i posted a comment with the failing block and verified it should be the same block therefore a real bug